### ISSUE TITLE
getImageBlob() segfault after using annotateImage() with WebP & imagemagick >= 7.0.7-12 

### ISSUE THREAD

**Issue by vrobert78** [ImageMagick/ImageMagick#913]
Hello

We've encountered a bug while using php / pecl imagick + annotateImage() followed by getImageBlob() on a WebP picture with imagemagick >= 7.0.7-12 resulting in a segfault.

We can use getImageBlob() on any WebP picture without problem unless we previously use annotateImage().
In this case getImageBlob() produce a segfault

Here is the stacktrace with valgrind

```
==1== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==1== Using Valgrind-3.10.0 and LibVEX; rerun with -h for copyright info
==1== Command: php /data/annotate.php
==1==
==1== Syscall param write(buf) points to uninitialised byte(s)
==1==    at 0x6F39C00: __write_nocancel (syscall-template.S:81)
==1==    by 0xD1B256F: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1B9DC3: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1B590A: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1B59BA: FcConfigBuildFonts (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1BF695: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xB85D886: LoadFontConfigFonts (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB85DF8C: IsTypeTreeInstantiated (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB85CE2C: GetTypeInfo (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB85CEAC: GetTypeInfoByFamily (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB6916B7: RenderType (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB691305: GetTypeMetrics (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==  Address 0xe4ec8a4 is 180 bytes inside a block of size 15,136 alloc'd
==1==    at 0x4C28C20: malloc (vg_replace_malloc.c:296)
==1==    by 0xD1B1CDC: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1B9DB0: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1B590A: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1B59BA: FcConfigBuildFonts (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xD1BF695: ??? (in /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0)
==1==    by 0xB85D886: LoadFontConfigFonts (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB85DF8C: IsTypeTreeInstantiated (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB85CE2C: GetTypeInfo (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB85CEAC: GetTypeInfoByFamily (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB6916B7: RenderType (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB691305: GetTypeMetrics (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==
==1== Conditional jump or move depends on uninitialised value(s)
==1==    at 0xB9AB665: SetImageProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB9ACC4E: WebPEncodeProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xCB00E88: WebPReportProgress (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED29A: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB03CB6: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED5EF: VP8EncAnalyze (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB012F9: WebPEncode (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xB9AD8B2: WriteWEBPImage (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB6E5452: WriteImage (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB69CFAA: ImageToBlob (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB33920F: MagickGetImageBlob (in /usr/local/lib/libMagickWand-7.Q8HDRI.so.4.0.0)
==1==    by 0xB035826: zim_imagick_getimageblob (in /usr/local/lib/php/extensions/no-debug-non-zts-20160303/imagick.so)
==1==
==1== Conditional jump or move depends on uninitialised value(s)
==1==    at 0x6EA8DCC: vfprintf (vfprintf.c:1642)
==1==    by 0x6ED0408: vsnprintf (vsnprintf.c:119)
==1==    by 0xB79E2E2: FormatLocaleStringList (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB79E3BD: FormatLocaleString (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB9AB6A8: SetImageProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB9ACC4E: WebPEncodeProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xCB00E88: WebPReportProgress (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED29A: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB03CB6: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED5EF: VP8EncAnalyze (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB012F9: WebPEncode (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xB9AD8B2: WriteWEBPImage (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==
==1== Use of uninitialised value of size 8
==1==    at 0xB9AB6DA: SetImageProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB9ACC4E: WebPEncodeProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xCB00E88: WebPReportProgress (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED29A: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB03CB6: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED5EF: VP8EncAnalyze (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB012F9: WebPEncode (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xB9AD8B2: WriteWEBPImage (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB6E5452: WriteImage (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB69CFAA: ImageToBlob (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB33920F: MagickGetImageBlob (in /usr/local/lib/libMagickWand-7.Q8HDRI.so.4.0.0)
==1==    by 0xB035826: zim_imagick_getimageblob (in /usr/local/lib/php/extensions/no-debug-non-zts-20160303/imagick.so)
==1==
==1== Jump to the invalid address stated on the next line
==1==    at 0x87E4: ???
==1==    by 0xB9ACC4E: WebPEncodeProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xCB00E88: WebPReportProgress (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED29A: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB03CB6: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED5EF: VP8EncAnalyze (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB012F9: WebPEncode (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xB9AD8B2: WriteWEBPImage (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB6E5452: WriteImage (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB69CFAA: ImageToBlob (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xB33920F: MagickGetImageBlob (in /usr/local/lib/libMagickWand-7.Q8HDRI.so.4.0.0)
==1==    by 0xB035826: zim_imagick_getimageblob (in /usr/local/lib/php/extensions/no-debug-non-zts-20160303/imagick.so)
==1==  Address 0x87e4 is not stack'd, malloc'd or (recently) free'd
==1==
==1== Invalid read of size 1
==1==    at 0xD3F5688: ??? (in /lib/x86_64-linux-gnu/libgcc_s.so.1)
==1==    by 0xD3F66F7: _Unwind_Backtrace (in /lib/x86_64-linux-gnu/libgcc_s.so.1)
==1==    by 0x6F53525: backtrace (backtrace.c:109)
==1==    by 0xDAE0934: nr_signal_tracer_common (util_signals.c:112)
==1==    by 0xDABD6F9: nr_php_fatal_signal_handler (php_minit.c:553)
==1==    by 0x742888F: ??? (in /lib/x86_64-linux-gnu/libpthread-2.19.so)
==1==    by 0x87E3: ???
==1==    by 0xB9ACC4E: WebPEncodeProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xCB00E88: WebPReportProgress (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED29A: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB03CB6: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED5EF: VP8EncAnalyze (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==  Address 0x87e4 is not stack'd, malloc'd or (recently) free'd
==1==
==1==
==1== Process terminating with default action of signal 11 (SIGSEGV)
==1==  Access not within mapped region at address 0x87E4
==1==    at 0xD3F5688: ??? (in /lib/x86_64-linux-gnu/libgcc_s.so.1)
==1==    by 0xD3F66F7: _Unwind_Backtrace (in /lib/x86_64-linux-gnu/libgcc_s.so.1)
==1==    by 0x6F53525: backtrace (backtrace.c:109)
==1==    by 0xDAE0934: nr_signal_tracer_common (util_signals.c:112)
==1==    by 0xDABD6F9: nr_php_fatal_signal_handler (php_minit.c:553)
==1==    by 0x742888F: ??? (in /lib/x86_64-linux-gnu/libpthread-2.19.so)
==1==    by 0x87E3: ???
==1==    by 0xB9ACC4E: WebPEncodeProgress (in /usr/local/lib/libMagickCore-7.Q8HDRI.so.4.0.0)
==1==    by 0xCB00E88: WebPReportProgress (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED29A: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCB03CB6: ??? (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==    by 0xCAED5EF: VP8EncAnalyze (in /usr/lib/x86_64-linux-gnu/libwebp.so.5.0.1)
==1==  If you believe this happened as a result of a stack
==1==  overflow in your program's main thread (unlikely but
==1==  possible), you can try to increase the size of the
==1==  main thread stack using the --main-stacksize= flag.
==1==  The main thread stack size used in this run was 8388608.
==1==
==1== HEAP SUMMARY:
==1==     in use at exit: 3,589,642 bytes in 23,685 blocks
==1==   total heap usage: 80,783 allocs, 57,098 frees, 32,810,698 bytes allocated
==1==
==1== LEAK SUMMARY:
==1==    definitely lost: 256 bytes in 1 blocks
==1==    indirectly lost: 32 bytes in 1 blocks
==1==      possibly lost: 1,368,768 bytes in 13,822 blocks
==1==    still reachable: 2,220,586 bytes in 9,861 blocks
==1==         suppressed: 0 bytes in 0 blocks
==1== Rerun with --leak-check=full to see details of leaked memory
==1==
==1== For counts of detected and suppressed errors, rerun with: -v
==1== Use --track-origins=yes to see where uninitialised values come from
==1== ERROR SUMMARY: 9 errors from 6 contexts (suppressed: 0 from 0)

valgrind: the 'impossible' happened:
   main(): signal was supposed to be fatal

host stacktrace:
==1==    at 0x380A48EF: show_sched_status_wrk (m_libcassert.c:319)

sched status:
  running_tid=1
```

PHP: 7.1.12-fpm-jessie (docker container) 
OPENJPEG_VERSION=2.3.0
IMAGEMAGICK_VERSION>=7.0.7-12
PECL_IMAGEMAGICK_VERSION=3.4.3

Libraries: libwebp-dev libjpeg62-turbo-dev libpng-dev libjxr-tools libjxr-dev libtiff5-dev msttcorefonts libfontconfig1-dev

Compilations flags: 
```
export CFLAGS="-mtune=haswell" \
 && ./configure \
    --with-webp \
    --with-libtiff \
    --with-jpeg=yes \
    --with-openjp2=yes \
    --with-quantum-depth=8 \
    --with-png=yes \
    --with-fontconfig=yes \
    --with-freetype=yes \
    LIBS=-lfontconfig \
    --with-windows-font-dir=/usr/share/fonts/truetype/msttcorefonts
```
Best regards,

Vincent

**Comment by vrobert78** [comment]
related to : https://github.com/mkoppanen/imagick/issues/223

**Comment by urban-warrior** [comment]
We fixed this problem a few days ago. Look for the ImageMagick 7.0.7-16 release within a day or two. Let us know if the problem persists with this release.

**Comment by vrobert78** [comment]
I confirm: problem solved with ImageMagick 7.0.7-16
